pkg/cloudprovider/types.go

```diff
@@ -57,7 +57,9 @@ type CloudProvider interface {
 	// Create launches a NodeClaim with the given resource requests and requirements and returns a hydrated
 	// NodeClaim back with resolved NodeClaim labels for the launched NodeClaim
 	Create(context.Context, *v1.NodeClaim) (*v1.NodeClaim, error)
-	// Delete removes a NodeClaim from the cloudprovider by its provider id
+	// Delete removes a NodeClaim from the cloudprovider by its provider id. Delete should return
+	// NodeClaimNotFoundError if the cloudProvider instance is terminated and nil if deletion has been triggered
+	// but the cloudProvider instance has not terminated
```

Comment:
Rather than saying "but the cloudprovider instance has not terminated", I'd say something like, "Returning nil from this function simply means that the Deletion call was successful. It's up to the CloudProvider to define when to return NotFound based on each one's respective termination flows. Karpenter will continue polling for Deletion until it returns a NodeClaimNotFound error

Reply:
++, that makes sense to me